src/pybind/mgr/cephadm/services/ingress.py

```diff
@@ -197,7 +197,10 @@ def keepalived_generate_config(
         hosts = sorted(list(set([host] + [str(d.hostname) for d in daemons])))
 
         # interface
-        bare_ip = str(spec.virtual_ip).split('/')[0]
+        if spec.virtual_ip is not None:
+            bare_ip = str(spec.virtual_ip).split('/')[0]
+        else:
+            bare_ip = str(spec.virtual_ips_list[0]).split('/')[0]
```

Comment:
```
cephadm/services/ingress.py:203: error: Value of type "Optional[List[str]]" is not indexable
```
we would need to check that `spec.virtual_ips_list` is not `None` here for mypy (fyi you can run this locally with `tox -e mypy` from the `src/pybind/mgr` directory).

Although I wonder if just taking the first ip is the correct approach regardless. Maybe we should just have `bare_ips` as a list and then add another layer to the following loop and check each ip against each interface rather than just checking each interface against whatever ip is first in the list? Will take a loser look at this later.

Reply:
Thanks for the tox tip. It indeed help me quite a bit. And thanks for the review.

I already did the few changes for the validation.  As for the bare_ip part. My first though was to check with one of the VIPs, as usually they are in the same subnet. So it should not matter which one to take. But maybe my assumption is false.  And I guess doing a new layer of the loop can make the logic cleaner.

I will be off tomorrow. I will look into this again later this week. Any input you may have until then is appreciated.

Reply:
I made the change regarding this looping logic.